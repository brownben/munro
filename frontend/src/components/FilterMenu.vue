<!--
  Filter Menu

  Holds all the input for the filter menu for the results table. Emits an event with the full selection criteria when a field is changed. No props are used as data only needs to come from it, no data needs passed to the component
-->

<template>
  <accordion-dropdown title="Filter">
    <text-input
      v-model="preferences.name"
      label="Name:"
      class="inline-block w-full md:w-1/2 md:pr-2"
      @input="onChange"
    />
    <text-input
      v-model="preferences.club"
      label="Club:"
      class="inline-block w-full md:w-1/2 md:pl-2"
      @input="onChange"
    />
    <number-input
      v-model.number="preferences.minAge"
      :min="0"
      :max="120"
      label="Min Age:"
      class="inline-block w-1/2 pr-2 md:w-1/4 md:-mb-4"
      @input="onChange"
    />
    <number-input
      v-model.number="preferences.maxAge"
      :min="0"
      :max="120"
      label="Max Age:"
      class="inline-block w-1/2 pl-2 md:w-1/4 md:-mb-4"
      @input="onChange"
    />
    <checkbox-input
      v-model="preferences.male"
      class="inline-block w-1/2 pr-2 text-center md:w-1/4"
      label="Male:"
      @input="onChange"
    />
    <checkbox-input
      v-model="preferences.female"
      class="inline-block w-1/2 pl-2 text-center md:w-1/4"
      label="Female:"
      @input="onChange"
    />
  </accordion-dropdown>
</template>

<script>
import CheckboxInput from './CheckboxInput'
import AccordionDropdown from './AccordionDropdown'
import TextInput from './TextInput'
import NumberInput from './NumberInput'

export default {
  components: {
    CheckboxInput,
    AccordionDropdown,
    TextInput,
    NumberInput,
  },

  data: () => ({
    preferences: {
      name: '',
      club: '',
      minAge: 0,
      maxAge: 100,
      male: true,
      female: true,
    },
  }),

  methods: {
    onChange: function () {
      this.$emit('changed', this.preferences)
    },
  },
}
</script>
