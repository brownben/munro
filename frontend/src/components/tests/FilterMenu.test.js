/*
  Filter Menu Component Unit Tests
*/

import { mount } from '@vue/test-utils'
import FilterMenu from '@/components/FilterMenu'

test('Is a Vue Instance', () => {
  const wrapper = mount(FilterMenu)
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = mount(FilterMenu)
  expect(wrapper.element).toMatchSnapshot()
})

test('Emits Correct Values', () => {
  const wrapper = mount(FilterMenu)

  // Change all the Values
  wrapper.setData({
    preferences: {
      name: 'Tester',
      club: 'IND',
      minAge: 15,
      maxAge: 50,
      male: false,
      female: true,
    },
  })
  wrapper.vm.onChange()
  expect(wrapper.emitted().changed.length).toBe(1)
  expect(wrapper.emittedByOrder()[0].args[0]).toEqual({
    name: 'Tester',
    club: 'IND',
    minAge: 15,
    maxAge: 50,
    male: false,
    female: true,
  })
})
