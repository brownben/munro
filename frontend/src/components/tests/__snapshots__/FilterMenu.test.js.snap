// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<div
  class="mt-3 mb-6 bg-white rounded-shape-xl my-shadow accordion"
>
  <div
    class="accordion-head"
  >
    <h2
      class="inline-block h-12 p-2 px-4 text-2xl select-none font-heading text-main"
    >
      
      Filter
    
    </h2>
     
    <svg
      class="float-right h-12 p-2 fill-current text-main"
      viewBox="0 0 24 24"
    >
      <path
        d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
      />
       
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
    </svg>
  </div>
   
  <div
    class="p-3 pt-0 pb-1 accordion-body"
    name="shrink"
    style="display: none;"
  >
    <div
      class="relative text-input top--4 inline-block w-full md:w-1/2 md:pr-2"
    >
      <label
        class="relative p-1 pb-0 mx-3 text-sm bg-white select-none font-heading text-main bottom--3"
      >
        Name:
      </label>
       
      <input
        class="w-full px-4 py-2 border outline-none border-main rounded-shape text-body"
        type="text"
      />
    </div>
     
    <div
      class="relative text-input top--4 inline-block w-full md:w-1/2 md:pl-2"
    >
      <label
        class="relative p-1 pb-0 mx-3 text-sm bg-white select-none font-heading text-main bottom--3"
      >
        Club:
      </label>
       
      <input
        class="w-full px-4 py-2 border outline-none border-main rounded-shape text-body"
        type="text"
      />
    </div>
     
    <div
      class="relative appearance-none text-input top--4 inline-block w-1/2 pr-2 md:w-1/4 md:-mb-4"
    >
      <label
        class="relative p-1 pb-0 mx-3 text-sm bg-white select-none font-heading text-main bottom--3"
      >
        Min Age:
      </label>
       
      <input
        class="w-full px-4 py-2 border outline-none appearance-none border-main rounded-shape text-body"
        max="120"
        min="0"
        type="number"
      />
    </div>
     
    <div
      class="relative appearance-none text-input top--4 inline-block w-1/2 pl-2 md:w-1/4 md:-mb-4"
    >
      <label
        class="relative p-1 pb-0 mx-3 text-sm bg-white select-none font-heading text-main bottom--3"
      >
        Max Age:
      </label>
       
      <input
        class="w-full px-4 py-2 border outline-none appearance-none border-main rounded-shape text-body"
        max="120"
        min="0"
        type="number"
      />
    </div>
     
    <div
      class="mt-3 checkbox-input inline-block w-1/2 pr-2 text-center md:w-1/4"
    >
      <label
        class="select-none text-md font-heading text-main"
      >
        
    Male:
    
        <input
          class="hidden"
          type="checkbox"
        />
         
        <span />
      </label>
    </div>
     
    <div
      class="mt-3 checkbox-input inline-block w-1/2 pl-2 text-center md:w-1/4"
    >
      <label
        class="select-none text-md font-heading text-main"
      >
        
    Female:
    
        <input
          class="hidden"
          type="checkbox"
        />
         
        <span />
      </label>
    </div>
  </div>
</div>
`;
